### ISSUE TITLE
compiling on Solaris 10 SPARC

### ISSUE THREAD

**Issue by roostercactus** [squash/sudosh2#20]
Hi, we're still using v1.0.2 on Solaris 10 SPARC.  I thought I'd try to upgrade to the latest master.zip to be able to use the wildcard in /etc/sudosh.conf to allow any remote command.  When I compile it, I'm getting this.  Any ideas?  

./configure --prefix=/usr/local --with-logdir=/opt/sudosh
checking build system type... sparc-sun-solaris2.10
checking host system type... sparc-sun-solaris2.10
checking target system type... sparc-sun-solaris2.10
checking for a BSD-compatible install... ./install-sh -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... nawk
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking to see if you want to log input too... no
checking for logdir... /opt/sudosh
checking for /etc/shells... found
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /usr/sfw/bin/ggrep
checking for egrep... /usr/sfw/bin/ggrep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for unistd.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking stdarg.h usability... yes
checking stdarg.h presence... yes
checking for stdarg.h... yes
checking stdio.h usability... yes
checking stdio.h presence... yes
checking for stdio.h... yes
checking sgtty.h usability... yes
checking sgtty.h presence... yes
checking for sgtty.h... yes
checking pty.h usability... no
checking pty.h presence... no
checking for pty.h... no
checking signal.h usability... yes
checking signal.h presence... yes
checking for signal.h... yes
checking time.h usability... yes
checking time.h presence... yes
checking for time.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking stropts.h usability... yes
checking stropts.h presence... yes
checking for stropts.h... yes
checking util.h usability... no
checking util.h presence... no
checking for util.h... no
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking sys/termios.h usability... yes
checking sys/termios.h presence... yes
checking for sys/termios.h... yes
checking for sys/types.h... (cached) yes
checking libutil.h usability... no
checking libutil.h presence... no
checking for libutil.h... no
checking features.h usability... no
checking features.h presence... no
checking for features.h... no
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking pwd.h usability... yes
checking pwd.h presence... yes
checking for pwd.h... yes
checking syslog.h usability... yes
checking syslog.h presence... yes
checking for syslog.h... yes
checking usersec.h usability... no
checking usersec.h presence... no
checking for usersec.h... no
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking for sys/stat.h... (cached) yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking ctype.h usability... yes
checking ctype.h presence... yes
checking for ctype.h... yes
checking for strings.h... (cached) yes
checking for openpty... no
checking for getusershell... yes
checking for gettimeofday... yes
checking for mkdir... yes
checking for rmdir... yes
checking for setpenv... no
checking for atoi... yes
checking for sscanf... yes
checking for open... yes
checking for read... yes
checking for close... yes
checking for fflush... yes
checking for fclose... yes
checking for fopen... yes
checking for fread... yes
checking for fputs... yes
checking for fprintf... yes
checking for strncpy... yes
checking for snprintf... yes
checking for strcmp... yes
checking for strlen... yes
checking for strftime... yes
checking for isspace... yes
checking for localtime... yes
checking for opendir... yes
checking for readdir... yes
checking for closedir... yes
checking for memchr... yes
checking for memmove... yes
checking for memset... yes
checking for select... yes
checking for strrchr... yes
checking for bzero... yes
checking for an ANSI C-conforming const... yes
checking for pid_t... yes
checking for size_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking vfork.h usability... no
checking vfork.h presence... no
checking for vfork.h... no
checking for fork... yes
checking for vfork... yes
checking for working fork... yes
checking for working vfork... (cached) yes
checking whether gcc needs -traditional... no
checking sys/select.h usability... yes
checking sys/select.h presence... yes
checking for sys/select.h... yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking types of arguments for select... int,fd_set *,struct timeval *
checking return type of signal handlers... void
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating src/sudosh.conf
config.status: creating sudosh.1
config.status: creating sudosh-replay.8
config.status: creating sudosh.conf.5
config.status: creating config.h
config.status: executing depfiles commands
# Configuration summary:

  Host type................: sparc-sun-solaris2.10
  CC.......................: gcc
  CFLAGS...................: -g -O2 -Werror

  Package..................: sudosh2
  Version..................: 1.0.6

  Installation prefix......: /usr/local
  Man directory............: /usr/local/share/man
  sysconfdir...............: /etc
  recording input..........: no

If this summary looks good, type 'make' or 'gmake' to compile
then 'make install' or 'gmake install' as root to install the
programs to the above installation prefix.

$ gmake
gmake  all-recursive
gmake[1]: Entering directory `/usr/homes/xxx/sudosh2-master'
Making all in src
gmake[2]: Entering directory`/usr/homes/xxx/sudosh2-master/src'
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT sudosh.o -MD -MP -MF .deps/sudosh.Tpo -c -o sudosh.o sudosh.c
mv -f .deps/sudosh.Tpo .deps/sudosh.Po
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT rand.o -MD -MP -MF .deps/rand.Tpo -c -o rand.o rand.c
mv -f .deps/rand.Tpo .deps/rand.Po
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT parse.o -MD -MP -MF .deps/parse.Tpo -c -o parse.o parse.c
mv -f .deps/parse.Tpo .deps/parse.Po
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT string.o -MD -MP -MF .deps/string.Tpo -c -o string.o string.c
mv -f .deps/string.Tpo .deps/string.Po
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT util.o -MD -MP -MF .deps/util.Tpo -c -o util.o util.c
mv -f .deps/util.Tpo .deps/util.Po
gcc  -g -O2 -Werror   -o sudosh sudosh.o rand.o parse.o string.o util.o  
gcc -DHAVE_CONFIG_H -I. -I..     -g -O2 -Werror -MT replay.o -MD -MP -MF .deps/replay.Tpo -c -o replay.o replay.c
replay.c: In function `main':
replay.c:173: error: structure has no member named`d_type'
replay.c:173: error: `DT_DIR' undeclared (first use in this function)
replay.c:173: error: (Each undeclared identifier is reported only once
replay.c:173: error: for each function it appears in.)
gmake[2]: *** [replay.o] Error 1
gmake[2]: Leaving directory`/usr/homes/xxx/sudosh2-master/src'
gmake[1]: **\* [all-recursive] Error 1
gmake[1]: Leaving directory `/usr/homes/xxx/sudosh2-master'
gmake: **\* [all] Error 2

This is on Solaris 10 Update 11 SPARC with the latest recommended patch bundle applied.  

$ gcc -v
Reading specs from /usr/sfw/lib/gcc/sparc-sun-solaris2.10/3.4.3/specs
Configured with: /sfw10/builds/build/sfw10-patch/usr/src/cmd/gcc/gcc-3.4.3/configure --prefix=/usr/sfw --with-as=/usr/ccs/bin/as --without-gnu-as --with-ld=/usr/ccs/bin/ld --without-gnu-ld --enable-languages=c,c++ --enable-shared
Thread model: posix
gcc version 3.4.3 (csl-sol210-3_4-branch+sol_rpath)

$ gmake -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for sparc-sun-solaris2.10

Thanks for looking.

**Comment by roostercactus** [comment]
Hi, I was able to compile v1.0.5 (last release on sourceforge) on Solaris 10 SPARC.  So while that's good, it doesn't support allowing wildcard commands in sudosh.conf.  

I ended up trying two things.  

One, I took the commit from issue #11 that edits sudosh.c and made the same edit on v1.0.5 sudosh.c.  This compiled fine, but didn't allow wildcards.  

Two, since the compile was failing on replay.c...I just replaced the replay.c file from master with the one from v1.0.5 and it compiled on Solaris 10 SPARC without error.  It seems to still work fine and also allows me to use the wildcard in sudosh.conf.  Sweet.  

Anyway, this is probably not a "good" idea but it looks like it works (in five minutes of basic functional testing) as a workaround for Solaris 10.

**Comment by squash** [comment]
I have a patch ready to go for this today. I don't have a solaris environment to test against but it is a simple patch addressing a difference between glibc and Solaris libc.
This and another issue have motivated me to start a rewrite as platforms contribute to drift apart and limit what new features might be added.
On Nov 16, 2016 1:38 PM, roostercactus <notifications@github.com> wrote:Hi, I was able to compile v1.0.5 (last release on sourceforge) on Solaris 10 SPARC.  So while that's good, it doesn't support allowing wildcard commands in sudosh.conf.  

I ended up trying two things.  

One, I took the commit from issue #11 that edits sudosh.c and made the same edit on v1.0.5 sudosh.c.  This compiled fine, but didn't allow wildcards.  

Two, since the compile was failing on replay.c...I just replaced the replay.c file from master with the one from v1.0.5 and it compiled on Solaris 10 SPARC without error.  It seems to still work fine and also allows me to use the wildcard in sudosh.conf.  Sweet.  

Anyway, this is probably not a "good" idea but it looks like it works (in five minutes of basic functional testing) as a workaround for Solaris 10.
